### ISSUE TITLE
Unable to delete the original variant 

### ISSUE THREAD

**Issue by bryanboza** [dotCMS/core#23573]
As part of the experiments, we are adding a original variant, but if you try to delete this one to replace for a new one, we are not able to do that, the notification says you was successfully deleted, but it was not  deleted.

### To Reproduce
- Login as admin
- Create a new page 
- Add a new variant 
- Try  to delete the original one

### Expected behavior
We have two ways here: 
- If we allow to delete it, we need to be able to delete 
- If delete the original is not an option, we need to lock the delete icon to be unable to click on it. 

### Screenshots
https://user-images.githubusercontent.com/2641437/207432132-6f008e99-fb71-4018-a850-d137368c658b.mov

### Desktop (please complete the following information):
Tested on release-23.01 // Docker // FF 

## Acceptance Criteria 

- We need to test it from UI
- Also need to test the endpoint

**Comment by dsilvam** [comment]
Passed Internal QA: Working as expected.

**Comment by bryanboza** [comment]
- The endpoint should return a forbidden code (403) instead 400 (bad request)

As part of this fix we need to add a postman test to validate the response code and error message here: 
### Call info

`DELETE: {{serverURL}}/api/v1/experiments/$experimentId/variants/$variantId`

### Test
```
 pm.test("Body matches string", function () {
        pm.response.to.have.status(401);
        pm.expect(pm.response.text()).to.include("Cannot delete Original Variant");
    });
```

**Comment by dsilvam** [comment]
We will respond with 405 when attempting to DELETE the original variant.

**Comment by spbolton** [comment]
@dsilvam Integration test is failing in master due to the new logic.  ExperimentAPIImpIT. testADeleteOriginalVariant_shouldFail is expecting and IllegalArgumentException but is now getting an NotAllowedException.

**Comment by bryanboza** [comment]
Fixed, tested on release-23.03 // Docker // FF
